ibm_quantum_schemas/models/executor/version_0_1/models.py

```diff
@@ -111,18 +117,25 @@ class SamplexItemModel(BaseModel):
     The non-trivial axes it introduces enumerate randomizations.
     """
 
-    chunk_size: int = Field(ge=1)
-    """The maximum number circuit arguments to bind to the circuit per shot loop."""
+    chunk_size: Union[Annotated[int, Field(ge=1)], Literal["auto"]]
+    """The maximum number circuit arguments to bind to the circuit per shot loop.
+
+    When ``"auto"``, the number will be chosen server-side with heuristics designed to optimize
+    execution speed. A quantum program must have items where either all chunk sizes are
+    integer-valued, or all chunk sizes are ``"auto"``.
+    """
 
     @model_validator(mode="after")
     def cross_validate(self) -> Self:
         """Check for mutual compatibility of types and shapes of attributes."""
         circuit = self.circuit.to_quantum_circuit(use_cached=True)
         samplex = self.samplex.to_samplex(use_cached=True)
 
-        outputs = samplex.outputs()
-        out_params = next(iter(spec for spec in outputs.specs if spec.name == "parameter_values"))
-        if (num_samplex_out := out_params.shape[-1]) != circuit.num_parameters:
+        if specs := samplex.outputs().get_specs("parameter_values"):
+            num_output_params = specs[0].shape[-1]
+        else:
+            num_output_params = 0
+        if (num_samplex_out := num_output_params) != circuit.num_parameters:
```

Comment:
This fixes a bug where empty circuits don't produce Samplexes with a spec for parameter_values, causing a StopIteration bug. Maybe build needs to be updated in samplomatic, do we ever want Samplexes built by build() without parameter_values?